src/lib/models/downloadRing/index.ts

```diff
@@ -0,0 +1,6 @@
+export type DownloadStatus =
```

Comment:
I think you might want to the status to the component itself. Maybe `DownloadRingDownloadStatus` would be more fitted?

Reply:
How about `DownloadRingStatus`?